packages/ui/src/components/editor/panels/Assets/container/index.tsx

```diff
@@ -473,8 +486,8 @@ const AssetPanel = () => {
     // TODO: add settings functionality
   }
 
-  const handleUpdateAsset = () => {
-    // TODO: add upload asset functionality
+  const refreshDirectory = async () => {
+    assetsQuery.refetch()
   }
```

Comment:
refreshDirectory is defined as an async arrow function, but there's no awaiting within the block context, maybe assetsQuery.refresh() is missing an await?